### ISSUE TITLE
Empty API Mock editor

### ISSUE THREAD

**Issue by dan-hook** [aws-amplify/amplify-category-api#1269]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v18.14.0

### Amplify CLI Version

10.7.0

### What operating system are you using?

Ubuntu 22.04 on WSL2 on Windows 10

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

Yes, the changes are extensive.

### Describe the bug

I ran apt-get install java-11-amazon-corretto-jdk to install Java.

Running amplify mock api launches Amplify GraphiQL Explore on localhost:20002, but it's empty saying "No Schema Available". 
 The CLI output is:
vscode ➜ .../panel-api/project/amplify/backend (features/VCD-504-ecv-api ✗) $ amplify mock api
GraphQL schema compiled successfully.

Edit your schema at /workspaces/gene-panel-api/project/amplify/backend/api/genepanelapi/schema.graphql or place .graphql files in a directory at /workspaces/gene-panel-api/project/amplify/backend/api/genepanelapi/schema
Mock does not handle CloudFormation resource of type AWS::IAM::ManagedPolicy. Skipping processing resource AuthRolePolicy01.
Mock does not handle CloudFormation resource of type AWS::IAM::ManagedPolicy. Skipping processing resource AuthRolePolicy02.
Mock does not handle CloudFormation resource of type AWS::IAM::ManagedPolicy. Skipping processing resource AuthRolePolicy03.
Data source of type RELATIONAL_DATABASE is not supported by local mocking. A NONE data source will be used.
Data source of type HTTP is not supported by local mocking. A NONE data source will be used.
Data source of type HTTP is not supported by local mocking. A NONE data source will be used.
Data source of type HTTP is not supported by local mocking. A NONE data source will be used.
Could not find ref for "RdsPanelDbName". Using unsubstituted value.
Could not find ref for "RdsPanelDbName". Using unsubstituted value.
Running GraphQL codegen
✔ Generated GraphQL operations successfully and saved at ../../src/graphql
AppSync Mock endpoint is running at http://172.17.0.2:20002
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Re-Trying with a new shard
Mock DB deletion detected. Clearing the OpenSearch indices...
Re-Trying with a new shard
Re-Trying with a new shard
Failed to start API Mocking. Running cleanup tasks.
MockProcessFault: Failed to start API Mocking.. Reason: null
    at APITest.start (/snapshot/repo/build/node_modules/amplify-util-mock/lib/api/api.js:151:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async start (/snapshot/repo/build/node_modules/amplify-util-mock/lib/api/index.js:18:5)
    at async Object.run (/snapshot/repo/build/node_modules/amplify-util-mock/lib/commands/mock/api.js:21:5)
    at async Object.executeAmplifyCommand (/snapshot/repo/build/node_modules/amplify-util-mock/lib/amplify-plugin-index.js:47:3)
    at async executePluginModuleCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:142:5)
    at async executeCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:40:9)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:117:5) {
  classification: 'FAULT',
  options: {
    message: 'Failed to start API Mocking.. Reason: null',
    link: 'https://docs.amplify.aws/cli/graphql/troubleshooting/'
  },
  downstreamException: undefined,
  toObject: [Function (anonymous)],
  details: undefined,
  resolution: undefined,
  code: undefined,
  link: 'https://docs.amplify.aws/cli/graphql/troubleshooting/'
}
### Expected behavior

I expected to see the API in the explorer.  

### Reproduction steps

Run 'amplify mock api'

### Project Identifier

9c19ea5526c1afd0a3e1987ea44f96af

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by josefaidt** [comment]
Hey folks I'm able to reproduce this issue using the Todo schema. Marking as a bug

**Comment by kmy90** [comment]
Hi, there is an ETA for the resolution of this bug?

**Comment by dwamianm** [comment]
@josefaidt. We are still experiencing this issue. Any update on the fix?

**Comment by espenbye** [comment]
We are experiencing the same issue on 11.0.5.

**Comment by dwamianm** [comment]
Checking for any updates on this bug. Still exists in 12.1.0

**Comment by remotehpc** [comment]
> Hey folks I'm able to reproduce this issue using the Todo schema. Marking as a bug ![image](https://user-images.githubusercontent.com/5033303/219821706-e0050a1b-a226-4f59-8ffd-41ac10dd53aa.png)

Hi Josef, are you able to resolve the bug or provide a walkaround? My team has been stuck here for too long, any help would be greatly appreciated! 

Best regards!

**Comment by aoi** [comment]
Hi josefaidt. 

Thank you for your response.
You have transfered [this issue](https://github.com/aws-amplify/amplify-category-api/issues/1646) to here.
But the code has probrem is not in this repository. It is in amplify-cli repository.

Please check my PR on amplify-cli repository. https://github.com/aws-amplify/amplify-cli/pull/12906/commits/ab4acb731c625b9578cbe8a9b90a8cc055dcf560

Best regards.

**Comment by aoi** [comment]
@josefaidt Hi josefaidt.

Thank you for your response.

You have transfered [this issue](https://github.com/aws-amplify/amplify-category-api/issues/1646) from amplify-cli repository to here.
But the code has probrem is not in this repository. It is in amplify-cli repository.
Please see my PR in amplify-cli repository.

amplify mock is very useful and important function for developers. A lot of developers are stuck in this bug for months.
I would appreciate it if you could respond promptly.

Best regards,

**Comment by pridapablo** [comment]
I found that the error directly comes from adding Cognito User Pool auth to the api, no matter if API or IAM is also enabled. Working with amplify -v 12.0.0 since amplify mocks fail completely on m2 Mac using the latest version of the cli.

**Comment by jmelendev** [comment]
Im using -v 12.2.0 and seeing the same issues. I am seeing the following error in the console:

"Error: Error when generating OIDC token: Right-hand side of 'instanceof' is not an object"

![Screen Shot 2023-07-26 at 9 35 59 PM](https://github.com/aws-amplify/amplify-category-api/assets/11486981/fdc6163a-7f21-476f-a000-b3d07cd331bb)

This was working fine earlier today, not sure what changed recently.

**Comment by araymond11** [comment]
Any updates ? Same issue with Amplify CLI version 12.3.0.

**Comment by kita3222** [comment]
I have confirmed that it also occurs in v12.4.0

**Comment by araymond11** [comment]
Still occurs in v12.7.0.

**Comment by afnx** [comment]
It's been months and still not able to see anything on Amplify GraphiQL Explorer. 

There is this `Uncaught (in promise) Error: Error when generating OIDC token: invalid 'instanceof' operand g` error that everybody mentioned on my browser console.

Also, when I try to execute a query, I always get the unauthorized error:

```console
{
  "errors": [
    {
      "errorType": "UnauthorizedException",
      "message": "UnauthorizedException: Missing authorization"
    }
  ]
}
```

**Comment by zakton5** [comment]
As mentioned by others developing and testing locally is very important and without the development process is far slower. @josefaidt Are there any updates?

**Comment by kmy90** [comment]
Hi everyone, this issue was opened almost 1 year ago. It is NOT a MINOR bug. There are teams around the world that are not able to use the latest version of Amplify just because of this problem. Could you help us, @josefaidt, @alharris-at ?

**Comment by rafadpedrosa** [comment]
I would like to share my experience, maybe will help someone. 

Yes, I'm having the same issue as everyone. But, given that I use nvm to control my node versions, I "manage" to deal with this problem installing different amplify-cli version on different node version as mentioned [here](https://github.com/aws-amplify/amplify-cli/issues/10488#issuecomment-1434523750). 

```
nvm use 20 
npm install npm install -g @aws-amplify

nvm use 18 
npm install -g @aws-amplify/cli@10.6.2 // working version
```

Also, I do agree with @kmy90. It should not be treated as a minor thing.

**Comment by afnx** [comment]
Hey folks, this is a serious issue here impacting everyone and disrupting the development process. Using workarounds, like downgrading the Amplify version, isn't the most convenient solution. Personally, I struggled with testing my projects locally using the explorer, and it really bothered me that such an important issue had been left unattended for a year.

I fixed it by creating a PR (https://github.com/aws-amplify/amplify-cli/pull/13555). Can you please support it? We need the Amplify team to review and approve the fix. Thanks!

**Comment by zepelega** [comment]
Still occurs in v12.10.1.

**Comment by afnx** [comment]
Fixed in version 12.11.0
